pkg/api/const.go

```diff
@@ -29,8 +29,9 @@ const (
 	Ubuntu1804      Distro = "ubuntu-18.04"
 	RHEL            Distro = "rhel"
 	CoreOS          Distro = "coreos"
-	AKS             Distro = "aks"
-	AKSDockerEngine Distro = "aks-docker-engine" // deprecated docker-engine distro
+	AKS             Distro = "aks"               // deprecated AKS 16.04 distro. Equivalent to aks-16.04.
```

Comment:
Rather than deprecating this, we could just call it a "mutable" reference. So at some point aks is redirected from 16.04 to 18.04

Reply:
so I rescind this, it's better to be precise and expressive about which OS version a distro is based upon

Reply:
yeah that was my initial thinking as well but after digging into the code I decided it was more clean to enforce explicit OS versioned distros.